### ISSUE TITLE
Add arrow key navigation to the inserter window.

### ISSUE THREAD

**Issue by BE-Webdesign** [WordPress/gutenberg#515]
In #462, having keyboard navigation for the inserter window is needed.

**Comment by aduth** [comment]
Related: #323

**Comment by BE-Webdesign** [comment]
In #462, Arrow key navigation is called for, any reason why tab navigation would be bad?

**Comment by jasmussen** [comment]
I don't know for sure, and I would imagine that @afercia might have thoughts. However, a couple of things to keep in mind:

1. down the road we could be looking at 50 maybe 100 blocks in the inserter, especially once the customization focus goes into full swing. 

2. in one scenario, when you invoke the inserter on a newline ("the newline is the commandline"), you are actually still typing on the newline as opposed to inside the inserter itself. Sort of like Slack slash commands:

![inserter](https://cloud.githubusercontent.com/assets/1204802/25473720/d26e3158-2b30-11e7-99b2-88b6fc7cfe96.gif)

☝️ those two points don't necessarily have to affect whether or not tab can work for navigating through items, just figured I'd give them as context for some of the behaviors we've been thinking of. The inserter as a whole is also one of the interface elements that will likely undergo more iteration as we test it.

**Comment by afercia** [comment]
Yep this kind of menus should work with both the Tab/Shift+Tab and Up/Down Arrow keys. Also, both tabbing and arrowing should be constrained within the inserter. Some good examples are on Twitter. Ideally, the inserter should reproduce their behavior:

<img width="409" alt="screen shot 2017-04-27 at 14 28 38" src="https://cloud.githubusercontent.com/assets/1682452/25483455/3681362a-2b56-11e7-8dae-306e4a4f7dbf.png">

The inserter case is a bit different, given the presence of the search field. I'd say both tabbing and arrowing should include the search field, but the actual interaction should be carefully tested so maybe better double check this point later.

Also, pressing Escape should close the inserter and move focus back in a proper place. Also clicking anywhere outside of the inserter should close it.

Worth noting when using screen readers, keyboard events need some special ARIA roles to actually work. Going to open a separate issue for this.

**Comment by afercia** [comment]
Please notice that arrows navigation should work also with the Left and Right arrow keys. That's the interaction some screen readers expect. For example, by default VoiceOver gives hints about which commands to use:

<img width="772" alt="arrows navigation also left and right" src="https://cloud.githubusercontent.com/assets/1682452/25493595/09f723e6-2b77-11e7-8b9f-9d72c352160e.png">

Seems to me that it would make sense to couple "Down" and "Right" to move to the next item and "Up" and "Left" to move to the previous one.
